internal/testing.go

```diff
@@ -3,17 +3,20 @@ package internal
 import (
 	"bytes"
 	"encoding/json"
+	"flag"
+	"fmt"
 	"io"
 	"log"
 	"net/http"
+	"os"
 	"reflect"
 )
 
 type (
 	Test struct {
-		Rule     io.Reader
```

Comment:
Is this change necessary?

Reply:
It makes it much easier to have the test failures print clearly what they failed on, like this:
![image](https://github.com/user-attachments/assets/c37aed5c-81cb-4504-9468-bf60be4199d1)